### PR TITLE
BAPI - byondapi test

### PR DESCRIPTION
**Pull request by DreamySkrell**
# what and why:
byondapi stuff
byondapi is cool
implementation is very heavily inspired by https://github.com/ParadiseSS13/Paradise/pull/25328
this PR adds a very minimal but usable rust project to the repo
this was test-merged to see if it actually works in game on the live server
but as it is now the PR does not add any byondapi calls to dm code

### CODE REVIEW

**Comment by DreamySkrell** [comment]
!review

**Comment by DreamySkrell** [comment]
!build_bapi

**Comment by DreamySkrell** [comment]
!build_bapi

**Comment by DreamySkrell** [comment]
!build_bapi

**Review by NonQueueingMatt — APPROVED** [review]

File: code/__DEFINES/rust_bapi.dm [hunk 1]
```diff
@@ -0,0 +1,39 @@
+
+// Default automatic .dll/.so detection.
+// Look for it in the build location first, then in `.`, then in standard places.
+
+/* This comment bypasses grep checks */ /var/__bapi
```
Comment (FluffyGhoster) [inline code/__DEFINES/rust_bapi.dm:5]: Why aren't we using GLOB here?

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**